src/dispatch/participant/flows.py

```diff
@@ -111,44 +117,34 @@ def remove_participant(user_email: str, incident: Incident, db_session: SessionL
 
 def reactivate_participant(user_email: str, incident: Incident, db_session: SessionLocal):
     """Reactivates a participant."""
-
-    # We get information about the individual
-    contact_plugin = plugin_service.get_active_instance(
-        db_session=db_session, project_id=incident.project.id, plugin_type="contact"
-    )
-    individual_info = contact_plugin.instance.get(user_email)
-    individual_fullname = individual_info["fullname"]
-
-    log.debug(f"Reactivating {individual_fullname} on incident {incident.name}...")
-
     participant = get_by_incident_id_and_email(
         db_session=db_session, incident_id=incident.id, email=user_email
     )
 
     if not participant:
-        log.debug(
-            f"{individual_fullname} is not an inactive participant of incident {incident.name}."
-        )
+        log.debug(f"{user_email} is not an inactive participant of {incident.name} incident.")
         return False
 
-    # We mark the participant as active
-    participant.is_active = True
+    log.debug(f"Reactivating {participant.individual.name} on {incident.name} incident...")
 
-    # We create a role for the participant
-    participant_role_in = ParticipantRoleCreate(role=ParticipantRoleType.participant)
+    # we get the last active role
+    participant_role = participant_role_service.get_last_active_role(
```

Comment:
I'm not sure this is what we want, if their last role was as incident commander and they leave and are re-added, I wouldn't want to give them that role again. I would imagine it would be simpler to just give them a participant role whenever they are reactivated.

Reply:
We don't allow incident commanders to leave the channel without first doing a handoff.